code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/IdentityExtension.java

```diff
@@ -13,232 +13,206 @@
-		if (event == null) {
-			return;
-		}
+	@Override
+	protected void onRegistered() {
+		// GENERIC_IDENTITY event listeners
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.GENERIC_IDENTITY,
+				IdentityConstants.EventSource.REQUEST_CONTENT,
+				this::handleEvent
+			);
 
-		cachedEvents.add(event);
-		processCachedEvents();
-	}
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.GENERIC_IDENTITY,
+				IdentityConstants.EventSource.REQUEST_RESET,
+				this::handleEvent
+			);
 
-	/**
-	 * Processes the cached events in the order they were received.
-	 */
-	void processCachedEvents() {
-		if (!state.hasBooted()) {
-			return;
-		}
+		// EDGE_IDENTITY event listeners
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.EDGE_IDENTITY,
+				IdentityConstants.EventSource.REQUEST_IDENTITY,
+				this::handleEvent
+			);
 
-		while (!cachedEvents.isEmpty()) {
-			final Event event = cachedEvents.peek();
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.EDGE_IDENTITY,
+				IdentityConstants.EventSource.UPDATE_IDENTITY,
+				this::handleEvent
+			);
 
-			if (EventUtils.isRequestIdentityEvent(event)) {
-				if (EventUtils.isGetUrlVariablesRequestEvent(event)) {
-					handleUrlVariablesRequest(event);
-				} else {
-					handleIdentityRequest(event);
-				}
-			} else if (EventUtils.isRequestContentEvent(event)) {
-				handleRequestContent(event);
-			} else if (EventUtils.isUpdateIdentityEvent(event)) {
-				handleUpdateIdentities(event);
-			} else if (EventUtils.isRemoveIdentityEvent(event)) {
-				handleRemoveIdentity(event);
-			} else if (EventUtils.isRequestResetEvent(event)) {
-				handleRequestReset(event);
-			} else if (EventUtils.isSharedStateUpdateFor(IdentityConstants.SharedState.IdentityDirect.NAME, event)) {
-				handleIdentityDirectECIDUpdate(event);
-			}
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.EDGE_IDENTITY,
+				IdentityConstants.EventSource.REMOVE_IDENTITY,
+				this::handleEvent
+			);
 
-			cachedEvents.poll();
-		}
+		// HUB shared state event listener
+		getApi()
+			.registerEventListener(
+				IdentityConstants.EventType.HUB,
+				IdentityConstants.EventSource.SHARED_STATE,
+				this::handleEvent
+			);
+	}
+
+	@Override
+	public boolean readyForEvent(@NonNull Event event) {
+		// Check if we are already booted
+		if (state.hasBooted()) return true;
+
+		// Attempt to boot
+		return state.bootupIfReady(sharedStateHandle);
 	}
 
 	/**
-	 * Calls {@link IdentityState#bootupIfReady(SharedStateCallback)} with a valid callback
-	 *
-	 * @return True if the bootup is complete
+	 * Responsible for handling the incoming events that this extension is registered for.
 	 */
-	boolean bootupIfReady() {
-		final SharedStateCallback callback = createSharedStateCallback();
-
-		return state.bootupIfReady(callback);
+	@VisibleForTesting
+	void handleEvent(@NonNull final Event event) {
```

Comment:
I like the swift version more where we use handlers per listener instead of one single entry point with if/elses, can we implement something like that?

Reply:
Yes, I agree that it will be cleaner that way! Will change it.